### PR TITLE
Add an option to display the specific results of too-late submissions for teams

### LINKED ISSUE #1967: Consider adding an option to choose whether or not to display the results of TOO-LATE submissions on the Team page
DOMjudge is mainly used as a competition system.

However, we often use DOMjudge for training contest, and then after the training contest, the team members who take part in the training contest have the need to solve problems after the contest.

It would be useful to add an option to the configuration page to enable the display of TOO-LATE submissions on the team page.

### PR DESCRIPTION
**Pull request by 2018hahazhufeng**
For team interface, show the specific judge results of too-late submissions. Meanwhile add a global configuration option to enable/disable this feature.
Closes #1967

### CODE REVIEW

File: etc/db-config.yaml [hunk 1]
```diff
@@ -215,6 +215,11 @@
             default_value: false
             public: true
             description: Should teams be able to view a diff of their and the reference output on sample testcases?
+        -   name: show_too-late_result
```
Comment (nickygerritsen) [inline etc/db-config.yaml:218]: I would call it 
```suggestion
        -   name: show_too_late_result
```

to be consistent with other option naming.

File: etc/db-config.yaml [hunk 2]
```diff
@@ -215,6 +215,11 @@
...
+            type: bool
+            default_value: false
+            public: true
+            description: Show specific results of TOO-LATE submissions in team interface?
```
Comment (nickygerritsen) [inline etc/db-config.yaml:222]: `specific` doesn't add much here, so:
```suggestion
            description: Show results of TOO-LATE submissions in team interface?
```

File: webapp/src/Controller/Team/SubmissionController.php [hunk 3]
```diff
@@ -114,6 +114,7 @@ public function viewAction(Request $request, int $submitId): Response
         $showCompile      = $this->config->get('show_compile');
         $showSampleOutput = $this->config->get('show_sample_output');
         $allowDownload    = (bool)$this->config->get('allow_team_submission_download');
+        $showToolateResult= $this->config->get('show_too-late_result');
```
Comment (nickygerritsen) [inline webapp/src/Controller/Team/SubmissionController.php:117]: Can we add a space before the `=`? (and realign the other stuff then)

**Review by nickygerritsen — APPROVED** [review]
That seemed easier than I would have expected 😄 .
Some minor comments but looks good.

File: webapp/src/Controller/Team/SubmissionController.php (hunk 3 above)
Comment (2018hahazhufeng) [reply webapp/src/Controller/Team/SubmissionController.php:117]: Okay I'll change it.

**Comment by nickygerritsen** [comment]
It's only some alignment for the whole table, right? In that case I wouldn't worry about it for now, we can fix it later.

**Comment by 2018hahazhufeng** [comment]
Yeah, the width of the result column is too big.

**Comment by nickygerritsen** [comment]
> Yeah, the width of the result column is too big.

OK I'm fine with ignoring this for now.

**Review by nickygerritsen — APPROVED** [review]
LGTM

File: webapp/templates/team/partials/submission_list.html.twig [hunk 4]
```diff
@@ -52,7 +52,16 @@
...
+                            {{ 'too-late' | printResult }} 
+                            {%- if showTooLateResult %}
+                                ({%- if submission.judgings.first is empty or submission.judgings.first.result is empty %}
+                                    {{- '' | printResult -}}
+                                {%- elseif verificationRequired and not submission.judgings.first.verified %}
+                                    {{- '' | printResult -}}
+                                {%- else %}
+                                    {{- submission.judgings.first.result | printResult -}}
+                                {%- endif %})
+                            {%- endif %}
```
Comment (vmcj) [inline webapp/templates/team/partials/submission_list.html.twig:64]: Is there a way to remove this duplication?

Comment (2018hahazhufeng) [reply webapp/templates/team/partials/submission_list.html.twig:64]: I've thought of this, but didn't find a better way to implement this.

The following code also deplicate some code but is shorter. Maybe it's better?
```
{%- if submission.submittime > submission.contest.endtime %}
    {{ 'too-late' | printResult }} 
{%- endif %}
{%- if submission.submittime <= submission.contest.endtime or showTooLateResult %}
    {%- if submission.submittime > submission.contest.endtime %}
        (
    {%- endif %}
    {%- if submission.judgings.first is empty or submission.judgings.first.result is empty %}
        {{- '' | printResult -}}
    {%- elseif verificationRequired and not submission.judgings.first.verified %}
        {{- '' | printResult -}}
    {%- else %}
        {{- submission.judgings.first.result | printResult -}}
    {%- endif %}
    {%- if submission.submittime > submission.contest.endtime -%}
        )
    {%- endif %}
{%- endif %}
```

Comment (vmcj) [reply webapp/templates/team/partials/submission_list.html.twig:64]: I like it a bit more. If you accept `too-late/<verdict>` you can also remove the last check.

Comment (2018hahazhufeng) [reply webapp/templates/team/partials/submission_list.html.twig]: Okay, I've made a commit of this.

**Review by vmcj — APPROVED** [review]

**Comment by meisterT** [comment]
Thanks for your contribution! Can you please rebase this on top of head so we can merge it?

**Comment by 2018hahazhufeng** [comment]
Sorry I might have misunderstood and squashed those commits just now

**Pull request merged**